### PR TITLE
PROD-30586: Hide blocked users in /all-members views.

### PR DESCRIPTION
**Pull request by denis-getopensocial**
## Problem

Users with Blocked status are displayed on the All Members page (`/all-members`).

## Solution

Update the views `/all-members` to hide blocked user.

## Issue tracker

https://getopensocial.atlassian.net/browse/PROD-30586

## How to test

- Create a user and set it blocked
- Access /all-members
- See that blocked users are not showing

## Release notes

Hide all blocked users from the view all members (/all-members).

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

File: modules/social_features/social_profile/social_profile.install [hunk 1]
```diff
@@ -202,3 +204,28 @@ function social_profile_update_121001(): void {
 function social_profile_update_130001(): void {
   user_role_grant_permissions('sitemanager', ['create terms in profile_tag']);
 }
+
+/**
+ * Update views /all-members to hide blocked members.
+ */
+function social_profile_update_130002(): void {
```
Comment (denis-getopensocial) [inline modules/social_features/social_profile/social_profile.install:211]: I picked 130002 so it runs, but I'm not so sure we want to want for 13.x to be released? Or do we?

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by BiaInacio** [comment]
Blocked users are not being shown on All members page!
image

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

File: modules/social_features/social_profile/social_profile.install (hunk 1 above)
Comment (robertragas) [reply modules/social_features/social_profile/social_profile.install:210]: 130002 is fine

File: modules/social_features/social_profile/social_profile.install [hunk 2]
```diff
@@ -202,3 +203,28 @@ function social_profile_update_121001(): void {
...
+  $moduleHandler = \Drupal::moduleHandler();
+  if ($moduleHandler->moduleExists('views') === FALSE ||
+    $moduleHandler->moduleExists('user') === FALSE ||
+    $moduleHandler->moduleExists('profile') === FALSE
+  ) {
+    return;
+  }
+
+  $config_file = \Drupal::service('extension.list.module')->getPath('social_profile') . '/config/install/views.view.newest_users.yml';
```
Comment (robertragas) [inline modules/social_features/social_profile/social_profile.install:219]: We should also copy the config and put it in /config/update/social_profile_update_130002.yml.
Reason for this is that the config in /install might change over time and we want to be sure that when the update hook runs it has the config it was tested for at that particular moment in time.

**Comment by robertragas** [comment]
Another remark is that the old view had multiple filters
Screenshot 2024-09-12 at 10 31 23
and the new one not anymore
Screenshot 2024-09-12 at 10 29 00
so this causes the admin to be visible there
Screenshot 2024-09-12 at 10 51 10

**Review by robertragas — CHANGES_REQUESTED** [review]
See remarks

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by denis-getopensocial** [comment]
So the issue here was the fact that views lost the current filters - so I added a filter, then removed it, then all the filters appeared. Then I re-exported the view and everything seems to be in order again.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4067-b59cqdjq0u0b9ulytdxek7gfhebfvj7h.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66def0b0e06d1f5785d54f2c

**Review by robertragas — APPROVED** [review]

**Comment by robertragas** [comment]
Everything fine now yes!

**Pull request merged**